lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/AttributeTranslationStrategy.php

```diff
@@ -176,7 +176,7 @@ public function getLocalesFor($document, NodeInterface $node, ClassMetadata $met
         $locales = array();
         foreach ($node->getProperties("*{$this->prefix}*") as $prop) {
             $matches = null;
-            if (preg_match('/' . $this->prefix . ':(..)-[^-]*/', $prop->getName(), $matches)) {
+            if (preg_match('/' . $this->prefix . ':([a-z]{2}_?[a-z]{0,2})-[^-]*/', $prop->getName(), $matches)) {
```

Comment:
to be BC, we need to accept `<any><any>` or `<any><any>_xx`. look at the parenthesis, the question mark includes the _. as i want to accept any character in the first group, i made the sublocalization one optional instead. not sure about making all lowercase, but found no clear indication - but here for example, the country is in uppercase: http://php.net/manual/de/intl.examples.basic.php